ForgeRock/forgerock-javascript-sdk
# [Sample App] Create a social login sample app in SDK

# JIRA Ticket

https://bugster.forgerock.org/jira/browse/SDKS-2538

# Description
[Sample App] Create a social login sample app in SDK
# Type of Change


# How Has This Been Tested?

**1) Local Log-in**

**STEPS TO TEST IT:**

- if not user, create a new user
- Click on Sign in
- Enter password and user, hit enter or click on Submit
- The page will check your credentials and, if successful, it will finish the log in flow.
- Go to Todo section
- create a new todo, should add it with no error
- Log out and log in, the previously todo should now appear on the list.

**2) Social-media log-in**

 **Pre- requisites:** The social identity provider service must be created and configured for the tenant. For this case, google social provider has been added as social identity provider service for the tenant only, via Access Management. 
**Important:** It may be necessary to clear site data before/after every log-in attempt
Next the configuration recommended for testing purposes:

AM_URL=https://openam-sdks.forgeblocks.com/am/
APP_URL=https://angular.example.com:8443
API_URL=http://localhost:9443
DEBUGGER_OFF=true
JOURNEY_LOGIN=google-login-web
JOURNEY_REGISTER=Registration
REALM_PATH=alpha
WEB_OAUTH_CLIENT=WebOAuthClient
PORT=9443
REALM_PATH=alpha
REST_OAUTH_SECRET=mysecret
REST_OAUTH_CLIENT=RestOAuthClient

**STEPS TO TEST IT:**

- Click on Sign in
- The page will provide as many options as  social identity provider services the tenants has configured 
- Click on Sign in with google
- Enter google credentials
- Enter ForgeRock credentials
- A notification will be sent to the Forge Rock authenticator App
- Approve notification
- The page will check your credentials and, if successful, it will finish the log in flow.
- Go to Todo section
- create a new todo, should add it with no error
- Log out and log in, the previously todo should now appear on the list.
[ForgeRock_ A Sample Angular (Web) Todo App.webm](https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/91f3619a-abaf-4973-8994-443ef753daec)

Next how the IdPs selection looks like

<img width="522" alt="Screenshot 2023-08-08 at 15 25 44" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/2a914ef9-6bfb-471a-b10c-4e14fc845e55">

# Definition of Done

Check all that apply

- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread
